sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1327,16 +1335,16 @@ def module_fw_upgrade(self, imagepath):
         imagepath specifies where firmware image file is located.
         target_firmware is a string that specifies the firmware version to upgrade to
 
-        This function returns True if download successfully completes. 
+        This function returns True if download successfully completes.
         Otherwise it will return False.
         """
         result = self.get_module_fw_info()
         try:
             _, _, _, _, _, _, _, _ = result['result']
         except (ValueError, TypeError):
             return result['status'], result['info']
-        result = self.get_module_fw_upgrade_feature()
-        try: 
+        result = self.get_module_fw_mgmt_feature()
+        try:
             startLPLsize, maxblocksize, lplonly_flag, autopaging_flag, writelength = result['result']
```

Comment:
get_module_fw_mgmt_feature() return changed the key from 'result' to 'feature'. Need to change here too.

Reply:
done. good catch.